tests/unit/test_project.py

```diff
@@ -426,33 +439,33 @@ def test_project_name_invalid(yaml_loaded_data, invalid_name):
 def test_project_missing_field(yaml_loaded_data, field):
     del yaml_loaded_data[field]
 
-    with pytest.raises(ProjectValidationError) as err:
-        Project.unmarshal(yaml_loaded_data)
+    with pytest.raises(CraftValidationError) as err:
+        load_project_yaml(yaml_loaded_data)
     assert str(err.value) == (
         "Bad rockcraft.yaml content:\n"
-        f"- field '{field}' required in top-level configuration"
+        f"- field {field} required in top-level configuration"
     )
 
 
 def test_project_extra_field(yaml_loaded_data):
     yaml_loaded_data["extra"] = "invalid"
 
-    with pytest.raises(ProjectValidationError) as err:
-        Project.unmarshal(yaml_loaded_data)
+    with pytest.raises(CraftValidationError) as err:
+        load_project_yaml(yaml_loaded_data)
     assert str(err.value) == (
         "Bad rockcraft.yaml content:\n"
-        "- extra field 'extra' not permitted in top-level configuration"
+        "- extra field extra not permitted in top-level configuration"
     )
 
 
 def test_project_parts_validation(yaml_loaded_data):
     yaml_loaded_data["parts"]["foo"]["invalid"] = True
 
-    with pytest.raises(ProjectValidationError) as err:
-        Project.unmarshal(yaml_loaded_data)
+    with pytest.raises(CraftValidationError) as err:
+        load_project_yaml(yaml_loaded_data)
     assert str(err.value) == (
         "Bad rockcraft.yaml content:\n"
-        "- extra field 'invalid' not permitted in 'parts.foo' configuration"
+        "- extra field invalid not permitted in parts.foo configuration"
```

Comment:
One issue that we should improve in craft-application is here: the original message (from rockcraft's version for `format_pydantic_errors()`) had single quotes around the yaml fields (`invalid` and `parts.foo`), which are now gone. I think they improve the clarity of the message

Reply:
Good point! https://github.com/canonical/craft-application/pull/157